### PR TITLE
fix: CTAS on multiple statements

### PR DESCRIPTION
**Pull request by betodealmeida**
### SUMMARY

The current behavior for running CTAS (create table as select) in SQL Lab is to check for **each statement** if it's a `SELECT`, and if so prepend the query with `CREATE TABLE $table_name AS $query`. For example, if we have this query:

```sql
SELECT 1 AS col;
```

And we run CTAS by passing the table name `my_table` we get this query:

```sql
CREATE TABLE my_table AS
SELECT 1 AS col;
```

The problem is that for a query with multiple statements we run CTAS **for each one**. For example:

```sql
SELECT 1 AS col;
SELECT 2 AS col;
```

When run with CTAS and `my_table` will produce these 2 queries:

```sql
CREATE TABLE my_table AS
SELECT 1 AS col;
CREATE TABLE my_table AS
SELECT 2 AS col;
```

The first query runs successfully, but the second fails because the table already exists.

This PR fixes the CTAS behavior, making it more consistent with how multiple statements work in SQL Lab. It changes SQL Lab so that:

1. A CTAS query can have multiple statements, as long as the last one is a `SELECT`. This allows user to pre-process the data before loading it into the table. A simple example in MySQL would be:

```sql
SET @value = 42;
SELECT @value AS foo;
```

When run in a CTAS, this will create a table with the column `foo` with a row with the value 42,  as expected.

In Hive, a common pattern is to write queries like this:

```sql
INSERT OVERWRITE staging_table ...;
SELECT * FROM staging_table WHERE ...;
```

Which should also work.

2. A CVAS (create view as select) query, on the other hand, can have a single statement and it MUST be a `SELECT`. Before, there was no differentiation between CTAS and CVAS (other than the query generated).

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TEST PLAN

Tested manually, added unit tests.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/sql_lab.py [hunk 1]
```diff
@@ -337,6 +331,32 @@ def execute_sql_statements(  # pylint: disable=too-many-arguments, too-many-loca
...
+        and query.ctas_method == CtasMethod.VIEW
+        and not parsed_query.is_valid_cvas()
+    ):
+        raise SqlLabException(
+            _(
+                "CVAS (create view as select) can only be run with a query with "
+                "a single SELECT statement. Please make sure your query has only "
+                "a SELECT statement. Then, try running your query again."
+            )
+        )
```
Comment (villebro) [inline superset/sql_lab.py:358]: As the error messages imply a deep understanding of how `is_valid_ctas()` and `is_valid_cvas()` work, I wonder if these should be moved into `ParsedQuery` as something like `assert_is_valid_ctas()` and `assert_is_valid_cvas()` and have the error messages there?

**Review by villebro — APPROVED** [review]
Thanks for a verbose description, made this very easy to review 👍 This will surely be appreciated by advanced SQL Lab users. Minor non-blocking comment, LGTM!

File: superset/sql_lab.py (hunk 1 above)
Comment (betodealmeida) [reply superset/sql_lab.py:358]: Ah, great point, @villebro! I'll move the exception message closer to the checks.

**Review by bkyryliuk — APPROVED** [review]

**Comment by amitmiran137** [comment]
What would be an example use case for using CTAS ?

**Comment by mistercrunch** [comment]
> What would be an example use case for using CTAS ?

If you mean "why materialize a dataset?", the main reason is to create a summary and/or richer dataset to speed subsequent queries against this snapshot

If you mean "why using the Superset CTAS feature over simply writing your own `CREATE TABLE {} AS`" statement, it has to do with administrator/policy control. It's possible in Superset to NOT give DML access, but allow people to effectively do this in a controlled way, and even force a target location/naming scheme for the CTAS-generated table

**Comment by bkyryliuk** [comment]
+1 to this comment, we are actively leveraging it
> > What would be an example use case for using CTAS ?
> 
> If you mean "why materialize a dataset?", the main reason is to create a summary and/or richer dataset to speed subsequent queries against this snapshot
> 
> If you mean "why using the Superset CTAS feature over simply writing your own `CREATE TABLE {} AS`" statement, it has to do with administrator/policy control. It's possible in Superset to NOT give DML access, but allow people to effectively do this in a controlled way, and even force a target location/naming scheme for the CTAS-generated table

**Pull request merged**